src/ui/components/Timeline/NonLoadingRegions.tsx

```diff
@@ -0,0 +1,31 @@
+import { useSelector } from "react-redux";
+import { getNonLoadingTimeRanges } from "ui/reducers/app";
+import { getZoomRegion } from "ui/reducers/timeline";
+import { TimeRange } from "ui/utils/app";
+
+const NonLoadingRegion = ({ range }: { range: TimeRange }) => {
+  const { endTime } = useSelector(getZoomRegion)!;
+  const { start, end } = range;
+  const style = {
+    left: `${(start / endTime) * 100}%`,
+    width: `${((end - start) / endTime) * 100}%`,
+  };
```

Comment:
Some small bit of confusion about our domain model here, but it relates to the duration question above.

Looks like this `left` positioning logic kind of assumes either that the zoom region's `startTime` is 0 or that ranges are always relative to a specific range. I guess maybe that's true since our `TimelineState` only has a single `zoomRegion` it's just confusing to me initially because I'm used to seeing view logic related to start/time times needing to explicitly account for both.

Reply:
Yeah `zoomRegion.startTime` is always 0 and `zoomRegion.endTime` is the duration of the recording. Definitely overdue for a refactor.